### PR TITLE
Auth fe

### PR DESCRIPTION
**Pull request by Rama-Qandeel**
i finished register and login  in front-end and tested with back-end 
server

### CODE REVIEW

File: back-end/controllers/users_controller.js [hunk 1]
```diff
@@ -4,27 +4,30 @@ const jwt = require("jsonwebtoken")
 const express = require('express');
 const app = express();
 require("dotenv").config()
-
-
+ 
+
```
Comment (Besslan) [inline back-end/controllers/users_controller.js]: remove the line

File: back-end/controllers/users_controller.js [hunk 2]
```diff
@@ -4,27 +4,30 @@ const jwt = require("jsonwebtoken")
 const express = require('express');
 const app = express();
 require("dotenv").config()
-
-
+
```
Comment (Besslan) [inline back-end/controllers/users_controller.js]: remove the white space but keep the line empty

File: back-end/controllers/users_controller.js [hunk 3]
```diff
@@ -4,27 +4,30 @@ const jwt = require("jsonwebtoken")
...
+ 
+    
 const register = async (req, res) => {
-    const { first_name, last_name, address, city, region, phone_number, email, password,
+    const {first_name,last_name, address, city, region, phone_number,
+         email, password,
         image_profile, payment_id, role_id, store_id } = req.body
     console.log("SALT", process.env.SALT);
     const hashedPassword = await bcrypt.hash(password, Number(process.env.SALT));
     console.log(hashedPassword);
```
Comment (Besslan) [inline back-end/controllers/users_controller.js]: remove the console log

File: back-end/controllers/users_controller.js [hunk 4]
```diff
@@ -4,27 +4,30 @@ const jwt = require("jsonwebtoken")
...
-
-
+ 
+    
 const register = async (req, res) => {
-    const { first_name, last_name, address, city, region, phone_number, email, password,
+    const {first_name,last_name, address, city, region, phone_number,
+         email, password,
         image_profile, payment_id, role_id, store_id } = req.body
     console.log("SALT", process.env.SALT);
```
Comment (Besslan) [inline back-end/controllers/users_controller.js]: remove the console log

File: back-end/controllers/users_controller.js [hunk 5]
```diff
@@ -4,27 +4,30 @@ const jwt = require("jsonwebtoken")
...
     const hashedPassword = await bcrypt.hash(password, Number(process.env.SALT));
     console.log(hashedPassword);
     const data = [first_name, last_name, role_id, address, city, region, phone_number, email, hashedPassword,
         image_profile, payment_id, store_id]
-    const query = `INSERT INTO users (first_name,last_name,role_id,address,city,region,phone_number,email,password,
-        image_profile,payment_id,store_id)
+        const query = `INSERT INTO users (first_name,last_name,role_id,address,city,region,phone_number,email,password,
+        image_profile,payment_id,store_id)
```
Comment (Besslan) [inline back-end/controllers/users_controller.js]: is there a reason why this is indented? remove the indention if not needed also remove the space at the end of the line

File: back-end/controllers/users_controller.js [hunk 6]
```diff
@@ -4,27 +4,30 @@ const jwt = require("jsonwebtoken")
...
-    const query = `INSERT INTO users (first_name,last_name,role_id,address,city,region,phone_number,email,password,
-        image_profile,payment_id,store_id)
+        const query = `INSERT INTO users (first_name,last_name,role_id,address,city,region,phone_number,email,password,
+        image_profile,payment_id,store_id) 
     VALUES (?,?,?,?,?,?,?,?,?,?,?,?) `
+
     connection.query(query, data, (err, results) => {
         if (err) {
+            
             console.log(err);
```
Comment (Besslan) [inline back-end/controllers/users_controller.js]: remove the empty line also using throw err would be more optimal here

File: back-end/controllers/users_controller.js [hunk 7]
```diff
@@ -4,27 +4,30 @@ const jwt = require("jsonwebtoken")
...
-        }
-        console.log(results);
-        res.json(results)
+        } 
+        console.log(results); 
+        res.json(results) 
     })
-}
+}
```
Comment (Besslan) [inline back-end/controllers/users_controller.js]: remove the extra spaces at the end of lines 26, 27, 28, 30

File: back-end/controllers/users_controller.js [hunk 8]
```diff
@@ -62,15 +65,15 @@ const login = (req, res) => {
                     expiresIn: process.env.TOKEN_EXPIRATION,
                 };
                 const token = jwt.sign(payload, process.env.SECRET, options);
-                console.log(token);
+                // console.log(token);
```
Comment (Besslan) [inline back-end/controllers/users_controller.js]: remove the commented console log

File: back-end/controllers/users_controller.js [hunk 9]
```diff
@@ -62,15 +65,15 @@ const login = (req, res) => {
...
-                // res.status(422);
-                res.json({ error: "Invalid login check your password" });
+                res.status(422);
+                res.json([]);
             }
         } else {
-            // res.status(404);
-            res.json({ error: "Invalid login check your email" });
+            res.status(404);
+            res.json([]);
```
Comment (Besslan) [inline back-end/controllers/users_controller.js]: Why are you returning an array?

File: back-end/routes/main-route.js [hunk 10]
```diff
@@ -21,7 +21,7 @@ mainRouter.delete('/order', deleteStore);
 mainRouter.post('/item', createItem);
 mainRouter.get('/item', getItems);
 mainRouter.delete('/item', deleteItem);
-
+
```
Comment (Besslan) [inline back-end/routes/main-route.js]: remove this line

File: front-end/src/App.css [hunk 11]
```diff
@@ -0,0 +1,21 @@
...
+  color: black !important;
+  background-color: #b3beca !important;
+}
+
+.PageHeading {
+  margin-top: 10px;
+  margin-bottom: 10px;
+  color: black !important;
+}
+.navbar-nav{
```
Comment (Besslan) [inline front-end/src/App.css]: add an empty line in between the classes

File: front-end/src/App.css [hunk 12]
```diff
@@ -0,0 +1,21 @@
...
+display: flex;
+
+}
+
+.nav-item{
+text-decoration: none;
+list-style: none;
+margin-left:10px ;
```
Comment (Besslan) [inline front-end/src/App.css]: indent the code in side the class

File: front-end/src/App.css [hunk 13]
```diff
@@ -0,0 +1,21 @@
...
+}
```
Comment (Besslan) [inline front-end/src/App.css]: add an empty line at the end of the file

File: front-end/src/components/Header.js [hunk 14]
```diff
@@ -0,0 +1,38 @@
...
+
+               
+            </ul>    
+
+          </div>    
+
+        </nav> <br />   
+            </div>
+        )
+
```
Comment (Besslan) [inline front-end/src/components/Header.js]: remove the empty lines that are present from like 7 to 35

File: front-end/src/components/Login.js [hunk 15]
```diff
@@ -0,0 +1,68 @@
...
+    const handleSubmit = (event) => {
+        const data = {
+          email: email,
+          password: password,
+        };
+        axios
+        .post("http://localhost:5000/login", data)
+        .then((response) => {
+          if (response.data) {
+            //   console.log(response.data);
```
Comment (Besslan) [inline front-end/src/components/Login.js]: remove the console log

File: front-end/src/components/Login.js [hunk 16]
```diff
@@ -0,0 +1,68 @@
...
+          placeholder="Password"
+          value={password}
+          onChange={handleChange}
+          required
+        />
+      </div>
+      <button onClick={handleSubmit}>Login</button>
+            </div>
+        )
+
```
Comment (Besslan) [inline front-end/src/components/Login.js]: remove the empty space here

File: front-end/src/components/Register.js [hunk 17]
```diff
@@ -0,0 +1,182 @@
+import React, { useState } from "react";
+import { BrowserRouter as Router , Route, Link } from "react-router-dom"; 
+
+import axios from "axios";
+const Register = (props) => {
```
Comment (Besslan) [inline front-end/src/components/Register.js]: move the `axios` import to line 3 and add an empty line in  between it and `Register`

File: back-end/controllers/users_controller.js (hunk 9 above)
Comment (Rama-Qandeel) [reply back-end/controllers/users_controller.js]: if email or password are incorrect return empty array  not jwt and in front end if response.data is empty array  Invalid User the user rewrite his email and password and cant redirect to home page

File: front-end/src/App.css [hunk 18]
```diff
@@ -0,0 +1,20 @@
+.navheader {
```
Comment (Besslan) [inline front-end/src/App.css]: change it to `.nav-header`

File: front-end/src/App.css [hunk 19]
```diff
@@ -0,0 +1,20 @@
...
+  margin-top: 10px;
+  color: black !important;
+  background-color: #b3beca !important;
```
Comment (Besslan) [inline front-end/src/App.css]: why are you using `!important`? The same review goes for the lines below

File: front-end/src/App.css [hunk 20]
```diff
@@ -0,0 +1,20 @@
...
+}
+
+.PageHeading {
```
Comment (Besslan) [inline front-end/src/App.css]: change it to `.page-heading`

File: front-end/src/App.css [hunk 21]
```diff
@@ -0,0 +1,20 @@
...
+  margin-top: 10px;
+  margin-bottom: 10px;
+  color: black !important;
+}
+.navbar-nav {
```
Comment (Besslan) [inline front-end/src/App.css]: add a space in between the classes

File: front-end/src/App.js [hunk 22]
```diff
@@ -1,17 +1,26 @@
...
-      <div>
-        <h1>HELLO WORLD</h1>
-      </div>
+     <Router>
+     <Header/>
+     <Route exact path="/login" render={props => <Login {...props} />} />
+      <Route exact path="/register" render={props => <Register {...props} />} />
+      <Route exact path="/home" render={props => <Home {...props} />} />
+
+
```
Comment (Besslan) [inline front-end/src/App.js]: remove the empty lines

File: back-end/controllers/users_controller.js [hunk 23]
```diff
@@ -1,90 +1,114 @@
...
+    email,
+    password,
+    image_profile,
+    payment_id,
+    role_id,
+    store_id,
+  } = req.body;
+  console.log("SALT", process.env.SALT);
+  const hashedPassword = await bcrypt.hash(password, Number(process.env.SALT));
+  console.log(hashedPassword);
```
Comment (Besslan) [inline back-end/controllers/users_controller.js]: Remove the console logs

File: back-end/controllers/users_controller.js [hunk 24]
```diff
@@ -1,90 +1,114 @@
...
-        } else {
-            // res.status(404);
-            res.json({ error: "Invalid login check your email" });
-        }
-    });
+  const { email, password } = req.body;
+  const data = [email, password];
+  connection.query(query, data, async (err, result) => {
+    if (err) throw err;
+    console.log("result :", result[0]);
```
Comment (Besslan) [inline back-end/controllers/users_controller.js]: remove the console log

File: back-end/controllers/users_controller.js [hunk 25]
```diff
@@ -1,90 +1,114 @@
...
+  ];
+  const query = `INSERT INTO users (first_name,last_name,role_id,address,city,region,phone_number,email,password,
+        image_profile,payment_id,store_id) 
+    VALUES (?,?,?,?,?,?,?,?,?,?,?,?) `;
+
+  connection.query(query, data, (err, results) => {
+    if (err) {
+      console.log(err);
+    }
+    console.log(results);
```
Comment (Besslan) [inline back-end/controllers/users_controller.js]: remove the console log

File: back-end/controllers/users_controller.js [hunk 26]
```diff
@@ -1,90 +1,114 @@
...
+          address,
+          city,
+          birhday,
+          phone_number,
+        };
+        const options = {
+          expiresIn: process.env.TOKEN_EXPIRATION,
+        };
+        const token = jwt.sign(payload, process.env.SECRET, options);
+        // console.log(token);
```
Comment (Besslan) [inline back-end/controllers/users_controller.js]: remove the commented console log

**Pull request merged**